db/seeds.rb

```diff
@@ -6,159 +6,54 @@
-IntraAlaskaLinehaul.import [:dist_mi, :weight_lbs, :rate, :effective], rates.intra_ak_linehauls, batch_size: 500
-
-puts 'Loading ZIP3 to ZIP3 distances from DTOD...'
-distances = CSV.read(Rails.root.join('db', 'seeds', 'zip3_dtod_output.csv'), { headers: false, col_sep: ' ' })
-DtodZip3Distance.import [:orig_zip3, :dest_zip3, :dist_mi], distances, { batch_size: 500, validate: false }
-
-puts 'Loading discounts from top TSP (by BVS) per channel...'
-if ENV['SEEDS_ENC_IV'].blank? || ENV['SEEDS_ENC_KEY'].blank?
-  STDERR.puts 'Cannot load encrypted discounts file; ensure that both SEEDS_ENC_IV and SEEDS_ENC_KEY are set in your environment (check .env file)'
-else
-  cipher = OpenSSL::Cipher::AES256.new :CBC
-  cipher.decrypt
-  cipher.iv = [ENV['SEEDS_ENC_IV']].pack('H*')
-  cipher.key = [ENV['SEEDS_ENC_KEY']].pack('H*')
-
-  # path, year, and tdl. Remember that rates take effect May 15, so the two
-  # TDLs before that date belong to the previous year
-  discount_files = [
-    [Rails.root.join('db', 'seeds', 'No 1 BVS Dom Discounts - Eff 1Oct2017.csv.enc'), 2017, 2],
-    [Rails.root.join('db', 'seeds', 'No 1 BVS Dom Discounts - Eff 1Jan2018.csv.enc'), 2017, 3]
-  ]
-
-  discount_files.each do |path, year, tdl|
-    discounts_csv = cipher.update(File.read(path))
-    discounts_csv << cipher.final
-
-    # after each discounts file, reset the cipher to read more files
-    cipher.reset
-
-    # date ranges for when the BVS scores from specific performance periods take effect
-    dates_by_tdl = [
-      (Date.new(year, 5, 15)..Date.new(year, 7, 31)),
-      (Date.new(year, 8, 1)..Date.new(year, 9, 30)),
-      (Date.new(year, 10, 1)..Date.new(year, 12, 31)),
-      (Date.new(year + 1, 1, 1)..Date.new(year + 1, 3, 6)),
-      (Date.new(year + 1, 3, 7)..Date.new(year + 1, 5, 14))
-    ]
-
-    tdl_daterange = dates_by_tdl[tdl]
-
-    discounts = CSV.parse(discounts_csv, headers: true)
-    discounts.each do |row|
-      TopTspByChannelLinehaulDiscount.where(orig: row['ORIGIN'], dest: row['DESTINATION'], tdl: tdl_daterange).first_or_initialize.tap do |d|
-        d.discount = row['LH_RATE']
-        d.save
-      end
-    end
-  end
-end
+puts '-- Seeding ZIP3s...'
+Seeds::Zip3s.new.seed!
+
+puts '-- Seeding ZIP5 rate areas...'
+Seeds::Zip5RateAreas.new.seed!
+
+puts '-- Seeding DTOD ZIP3 to ZIP3 distances...'
+Seeds::DtodZip3Distances.new.seed!
+
+puts '-- Seeding 2017 400NG baseline rates...'
+Seeds::BaselineRates.new(
+  date_range: Range.new(Date.parse('2017-05-15'), Date.parse('2018-05-14')),
+  file_path: Rails.root.join('lib', 'data', '2017 400NG Baseline Rates.xlsx')
+).seed!
+
+puts '-- Seeding 2018 400NG baseline rates...'
+Seeds::BaselineRates.new(
+  date_range: Range.new(Date.parse('2018-05-15'), Date.parse('2019-05-14')),
+  file_path: Rails.root.join('lib', 'data', '2018 400NG Baseline Rates.xlsx')
+).seed!
+
+puts '-- Seeding top TSP by channel linehaul discounts...'
+Seeds::TopTspByChannelLinehaulDiscounts.new.seed!
```

Comment:
Most of these commands should be source order independent with the exception of the `ShippingOffice` and `TransportationOffice` imports. Shipping Offices should be imported first so that Transportation Offices can be correctly associated with their Shipping Office.